docarray/array/storage/memory/seqlike.py

```diff
@@ -66,7 +67,5 @@ def __add__(self, other: Union['Document', Iterable['Document']]):
         return v
 
     def extend(self, values: Iterable['Document']) -> None:
-        values = list(values)  # consume the iterator only once
-        last_idx = len(self._id2offset)
-        self._data.extend(values)
-        self._id_to_index.update({d.id: i + last_idx for i, d in enumerate(values)})
+        for v in values:
+            self._data[v.id] = v
```

Comment:
another option is:
```
values = list(values)
tmp = pd.Series(values, index=(d.id for d in values))
self._data = self._data.append(tmp)
```

Otherwise, there is no need to implement `extend`, it will automatically call `append()`/ `insert()`, simply remove it